src/Microsoft.Health.Dicom.Core/Features/Store/StoreValidatorResultBuilder.cs

```diff
@@ -0,0 +1,89 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.ObjectModel;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.Store;
+
+internal sealed class StoreValidatorResultBuilder
+{
+    public StoreValidatorResultBuilder()
+    {
+        ErrorMessages = new Collection<string>();
+        WarningMessages = new Collection<string>();
+        WarningCodes = ValidationWarnings.None;
+        FirstException = null;
+    }
+
+    private Collection<string> ErrorMessages { get; }
+
+    private Collection<string> WarningMessages { get; }
+
+    private ValidationWarnings WarningCodes { get; set; }
+
+    // TODO: Remove this during the cleanup. *** Hack to support the existing validator behavior ***
+    private Exception FirstException { get; set; }
+
+    public StoreValidatorResult Build()
+    {
+        return new StoreValidatorResult(
+            ErrorMessages,
+            WarningMessages,
+            WarningCodes,
+            FirstException);
+    }
+
+    public void AddError(Exception ex, QueryTag queryTag = null)
```

Comment:
I'm on the fence on these methods. A few questions:
1. Is there a single `StoreValidatorResult` per entry in the STOW request or one per STOW request?
2. Should we be recording the mapping of query tag -> exception(s)?

Reply:
1) One per STOW request
2) My initial thought was to implement this as K, V (QueryTag vs. Error/Warning/Exception). But we deferred this to a later phase.

Reply:
Can we reduce the APIs to simply `public StoreValidatorResultBuilder AddError(Exception exception)` and `public StoreValidatorResultBuilder AddWarning(Exception exception)` (or perhaps `string` for warnings)? We could perhaps simply wrap messages in a generic validation exception.